browser/brave_ads/ads_tab_helper.cc

```diff
@@ -136,7 +136,12 @@ void AdsTabHelper::DocumentOnLoadCompletedInPrimaryMainFrame() {
     return;
   }
 
-  RunIsolatedJavaScript(web_contents()->GetMainFrame());
+  content::RenderFrameHost* render_frame_host = web_contents()->GetMainFrame();
+  RunIsolatedJavaScript(render_frame_host);
+
+  if (ads_service_) {
+    ads_service_->MaybeRetrieveSearchResultAd(render_frame_host);
```

Comment:
why are we passing this through AdsService? We want to eliminate content dependencies on AdsService and this is adding more of them and further bloating AdsService. AdsTabHelper can own the handler

Reply:
made SearchResultAdService as a separate keyed service